src/Registry.sol

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import "openzeppelin-contracts/contracts/access/Ownable.sol";
+
+/// @title  Registry contract :: Checking if a token's legit
+/// @notice This contract can be called by our main contract for checking if the ERC20 token address is legit
+
+contract Registry is Ownable {
+    mapping(address => bool) isLegit;
+
+    event TokenSetTo(address indexed, bool value);
+
+    /// @notice To be called by the DAO, this switches ERC20 address in the isLegit mapping
+    /// @param tokenAddress ERC20 token address
+    /// @param _bool The value we want to switch to
+
+    function setIsLegit(address tokenAddress, bool _bool) public onlyOwner {
+        if (_bool == true) {
+            isLegit[tokenAddress] = true;
+            emit TokenSetTo(tokenAddress, true);
+        } else {
+            delete isLegit[tokenAddress];
+            emit TokenSetTo(tokenAddress, false);
+        }
```

Comment:
you can simplify this with:

```solidity
isLegit[tokenAddress] = _verified;
emit TokenSetTo(tokenAddress, _verified);
```

no need for if statement, `delete isLegit[tokenAddress]` is equivalent to `isLegit[tokenAddress] = false`